roles/ceph-rgw/tasks/multisite/secondary_connection_checks.yml

```diff
@@ -0,0 +1,47 @@
+  - name: convert period_pull to json
+    set_fact:
+      period_pull_json: "{{ period_pull.stdout | from_json }}"
+
+  - name: get zonegroups list from json_output
+    set_fact:
+      zonegroups: "{{ period_pull_json['period_map']['zonegroups'] }}"
+
+  - name: set zonegroups_endpoints list as an empty list
+    set_fact:
+      zonegroups_endpoints: []
+
+  - name: get zones from zonegroups list
+    set_fact:
+      zonegroups_endpoints: "{{ zonegroups_endpoints }} + {{ item.endpoints }}"
+    with_items: "{{ zonegroups }}"
```

Comment:
because `zonegroups` is a dict you should use `with_dict`
This means you also need to change `item.endpoints` by `item.value.endpoints`